app/api/data.py

```diff
@@ -322,13 +284,19 @@ def edit_core_data_from_states_daily():
             'No state specified in batch edit context', 'edit_core_data_from_states_daily')
         return flask.jsonify('No state specified in batch edit context'), 400
 
+    core_data = payload['coreData']
+    return edit_states_daily_internal(
+        get_jwt_identity(), context, core_data, state_to_edit, publish=True)
+
+def edit_states_daily_internal(user, context, core_data, state_to_edit=None, publish=False):
     flask.current_app.logger.info('Creating new batch from context: %s' % context)
 
     batch = Batch(**context)
-    batch.user = get_jwt_identity()
-    batch.isRevision = True
-    batch.isPublished = True  # edit batches are published by default
-    batch.publishedAt = datetime.utcnow()
+    batch.user = user
+    batch.isRevision = publish
```

Comment:
why is `isRevision` based on the value of `publish`?

Reply:
I don't know what `isRevision` means, it's not explained, but it looks like it always follows publish (and especially in these 2 methods).

Reply:
Ah sorry - it's meant to denote whether a particular row is an edit (as opposed to being written through WS2), so anything going in through this API endpoint should be `isRevision = True`.

Originally we assumed it would only be true in cases where there was existing data for that date/state, but now that rows can be added through the edit tool this isn't necessarily true.

Reply:
So what's the meaning of this field now? and in 2 weeks when everything is done through the API?

Reply:
The meaning hasn't changed (it should still be true for rows coming in through the edit endpoint), just our assumption around it being true only in the case of there being an existing corresponding row. If you set it to True here, it won't break correctness.

Reply:
So revision = "it edits previous data"?
It's not going to be correct to assign it to a full batch, since it can be only edit/edit-new mixed/only-new, all options are valid.

Or just "it came through one of the `edit` endpoints and not `insert`?

Reply:
I think of it as the latter meaning. And our "public" data read methods do the right thing whether a data point comes in the insert or the edit endpoints, so if you set it to true in this method, it won't break anything.